[BUG] Vertica Spark Connector cannot run on EMR

## Environment
- Spark version: Spark 3.1.2
- Hadoop version: Amazon 3.2.1 
- Vertica version: 11.1.
- Vertica Spark Connector version: 3.1.0
- Java version: openjdk version "1.8.0_332"
- Additional Environment Information: Running on EMR (emr-6.5.0)

---

## Problem Description
- Describe the issue in as much details as possible, so it is possible to reproduce it.

Vertica Spark Connector fails to run on EMR because the spark version resolves to `3.1.2-amzn-1`

The below code fails to parse the version (fails to convert the semantic version [patch] to integer), as a result considers that it is running on latest version and try to load (class=org.apache.spark.sql.connector.read.SupportsPushDownAggregates) which isnt avialable in 3.1.2 version.

https://github.com/vertica/spark-connector/blob/main/connector/src/main/scala/com/vertica/spark/datasource/v2/VerticaDatasourceV2Table.scala
```
var sparkNewerThan31 = true
        try {
          val sparkVersion = SparkSession.getActiveSession.get.version
          val versionList = sparkVersion.split("\\.").map(_.toInt)
          if(versionList.length >= 2 && versionList(0) == 3 && versionList(1) < 2) {
            sparkNewerThan31 = false
          }
        }
        catch { // Couldn't recgonize version string, assume newer version
          case _: java.lang.NumberFormatException => sparkNewerThan31 = true
        }
```
1. Steps to reproduce:
2. Expected behaviour:
3. Actual behaviour:
4. Error message/stack trace:
5. Code sample or example on how to reproduce the issue:

---

## Spark Connector Logs
- Add related logs entries here.

```java.lang.NoClassDefFoundError: org/apache/spark/sql/connector/read/SupportsPushDownAggregates
  at java.lang.ClassLoader.defineClass1(Native Method)
  at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
  at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
  at java.net.URLClassLoader.defineClass(URLClassLoader.java:473)
  at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
  at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
  at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
  at java.security.AccessController.doPrivileged(Native Method)
  at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
  at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
  at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
  at com.vertica.spark.datasource.v2.VerticaTable.newScanBuilder(VerticaDatasourceV2Table.scala:105)
  at com.vertica.spark.datasource.v2.VerticaTable.schema(VerticaDatasourceV2Table.scala:59)
  at org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation$.create(DataSourceV2Relation.scala:178)
  at org.apache.spark.sql.DataFrameReader.$anonfun$load$1(DataFrameReader.scala:303)
  at scala.Option.map(Option.scala:230)
  at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:265)
  at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:225)
  ... 47 elided
Caused by: java.lang.ClassNotFoundException: org.apache.spark.sql.connector.read.SupportsPushDownAggregates
  at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
  at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
  at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
  ... 65 more```

--- a/connector/src/main/scala/com/vertica/spark/datasource/v2/VerticaDatasourceV2Table.scala
+++ b/connector/src/main/scala/com/vertica/spark/datasource/v2/VerticaDatasourceV2Table.scala
@@ -88,12 +88,13 @@ class VerticaTable(caseInsensitiveStringMap: CaseInsensitiveStringMap, readSetup
         }
         logger.debug("Config loaded")
 
-        // Pushdown of aggregates added in spark 3.2, detect spark version so we return class with right feature support
+        // Aggregates push down were added in spark 3.2. We detect spark version here to return the compatible class
         var sparkNewerThan31 = true
         try {
-          val sparkVersion = SparkSession.getActiveSession.get.version
-          val versionList = sparkVersion.split("\\.").map(_.toInt)
-          if(versionList.length >= 2 && versionList(0) == 3 && versionList(1) < 2) {
+          val versionList = SparkSession.getActiveSession.get.version.split("\\.")
+          val major = versionList(0).toInt
+          val minor = versionList(1).toInt
+          if(versionList.length >= 2 && major == 3 && minor < 2) {
             sparkNewerThan31 = false
           }
         }

--- a/connector/src/main/scala/com/vertica/spark/datasource/v2/VerticaDatasourceV2Table.scala
+++ b/connector/src/main/scala/com/vertica/spark/datasource/v2/VerticaDatasourceV2Table.scala
@@ -92,10 +92,12 @@ class VerticaTable(caseInsensitiveStringMap: CaseInsensitiveStringMap, readSetup
         var sparkNewerThan31 = true
         try {
           val versionList = SparkSession.getActiveSession.get.version.split("\\.")
-          val major = versionList(0).toInt
-          val minor = versionList(1).toInt
-          if(versionList.length >= 2 && major == 3 && minor < 2) {
-            sparkNewerThan31 = false
+          if (versionList.length >= 2) {
+            val major = versionList(0).toInt
+            val minor = versionList(1).toInt
+            if (major == 3 && minor < 2) {
+              sparkNewerThan31 = false
+            }
           }
         }
         catch { // Couldn't recgonize version string, assume newer version

